packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -25,72 +20,73 @@ export function assetImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
     name: 'vite:asset-import-meta-url',
     async transform(code, id, options) {
       if (
-        !options?.ssr &&
-        code.includes('new URL') &&
-        code.includes(`import.meta.url`)
+        options?.ssr ||
+        !code.includes('new URL') ||
+        !code.includes(`import.meta.url`)
       ) {
-        const importMetaUrlRE =
-          /\bnew\s+URL\s*\(\s*('[^']+'|"[^"]+"|`[^`]+`)\s*,\s*import\.meta\.url\s*,?\s*\)/g
-        const noCommentsCode = code
-          .replace(multilineCommentsRE, blankReplacer)
-          .replace(singlelineCommentsRE, blankReplacer)
-          .replace(stringsRE, (m) => `'${'\0'.repeat(m.length - 2)}'`)
-
-        let s: MagicString | null = null
-        let match: RegExpExecArray | null
-        while ((match = importMetaUrlRE.exec(noCommentsCode))) {
-          const { 0: exp, 1: emptyUrl, index } = match
+        return null
+      }
+      let s: MagicString | undefined
+      let match: RegExpExecArray | null
+      const assetImportMetaUrlRE =
+        /\bnew\s+URL\s*\(\s*('[^']+'|"[^"]+"|`[^`]+`)\s*,\s*import\.meta\.url\s*,?\s*\)/g
```

Comment:
Normally you can extract these regexes out as you did here https://github.com/vitejs/vite/pull/7650/commits/be41a44c9c922170c685fe6d0c43525a485a9ff9#diff-300d1d46aa4bd9f65c9df6eb138c05c2b526a5b33c0a3a10ca9cad3173968a92R8,
but you need to call `.lastIndex = 0` on the regex before each iteration. You can check here:
https://github.com/vitejs/vite/blob/71868579058512b51991718655e089a78b99d39c/packages/vite/src/client/overlay.ts#L126

But I don't think we can do it here, as two requests may be being processed in parallel. We should maybe review the other places we do it to see if aren't hit by race conditions there. If you still want to take it out, you could do it as a regex factory so we have a new one each time (does the concept of a regex pool exists? 😅, just kidding, I don't think we win much optimizing these aways)

Reply:
maybe string template don't need to replace all. https://github.com/vitejs/vite/issues/7647
just need to 

```ts
const str = `${a${`c${`d`}`}b}`

const clean = `${\0${`\0${`\0`}`}\0}`
```

Reply:
oh, good trick.  I finally understand why there are doesn't work 🙈

> I don't think we can do it here, as two requests may be being processed in parallel

I think so.